File: java/parser/src/main/java/gov/lbl/parser/parser/Logical_factorVisitor.java
package gov.lbl.parser.parser;

import gov.lbl.antlr4.visitor.modelicaBaseVisitor;
import gov.lbl.antlr4.visitor.modelicaParser;
import gov.lbl.parser.domain.Logical_factor;
import gov.lbl.parser.domain.Relation;

public class Logical_factorVisitor extends modelicaBaseVisitor<Logical_factor> {
    @Override
    public Logical_factor visitLogical_factor(modelicaParser.Logical_factorContext ctx) {
        Boolean not = ctx.NOT() == null ? false : true;
        RelationVisitor relationVisitor = new RelationVisitor();
        Relation relation = ctx.relation() == null ? null : ctx.relation().accept(relationVisitor);
        return new Logical_factor(not, relation);
    }
}

File: jsParser/parser/Logical_factorVisitor.js
const modelicaVisitor = require('../antlrFiles/modelicaVisitor').modelicaVisitor
const Logical_factor = require('../domain/Logical_factor')

const RelationVisitor = require('./RelationVisitor')

class Logical_factorVisitor {
  constructor () {
    modelicaVisitor.call(this)
    return this
  }

  visitLogical_factor (ctx) {
    const not = !!ctx.NOT()
    let relation = null

    if (ctx.relation()) {
      const relationVisitor = new RelationVisitor.RelationVisitor()
      relation = relationVisitor.visitRelation(ctx.relation())
    }

    return new Logical_factor.Logical_factor(not, relation)
  }
};

Logical_factorVisitor.prototype = Object.create(modelicaVisitor.prototype)

exports.visitLogical_factor = this.visitLogical_factor
exports.Logical_factorVisitor = Logical_factorVisitor
